src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/DoNotGuardSetAddOrRemoveByContainsTests.cs

```diff
@@ -0,0 +1,1159 @@
+            {
+                mySetLocal4.Remove(""Item"");
+            }
+        }
+
+        private void RemoveItem(HashSet<string> setParam)
+        {
+            if (setParam.Contains(""Item""))
+                SetField1.Remove(""Item"");
+
+            if (!SetField1.Contains(""Item""))
+            {
+                setParam.Remove(""Item"");
+            }
+        }" + CSNamespaceAndClassEnd;
+
+            await VerifyCS.VerifyAnalyzerAsync(source);
+        }
+
+        [Fact]
+        public async Task ContainsAndAddCalledWithDifferentArguments_NoDiagnostic_CS()
+        {
+            string source = CSUsings + CSNamespaceAndClassStart + @"
+        private readonly HashSet<string> MySet = new HashSet<string>();
+        private const string OtherItemField = ""Other Item"";
+
+        public string OtherItemProperty { get; } = ""Other Item"";
+
+        public MyClass(string otherItemParameter)
+        {
+            if (!MySet.Contains(""Item""))
+                MySet.Add(""Other Item"");
+
+            if (!MySet.Contains(""Item""))
+                MySet.Add(otherItemParameter);
+
+            if (!MySet.Contains(""Item""))
+                MySet.Add(OtherItemField);
+
+            if (!MySet.Contains(""Item""))
+                MySet.Add(OtherItemProperty);
+
+            string otherItemLocal = ""Other Item"";
+            if (!MySet.Contains(""Item""))
+                MySet.Add(otherItemLocal);
+        }" + CSNamespaceAndClassEnd;
+
+            await VerifyCS.VerifyCodeFixAsync(source, source);
+        }
+
+        [Fact]
+        public async Task ContainsAndAddCalledWithSameArgumentsFields_OffersFixer_CS()
+        {
+            string source = CSUsings + CSNamespaceAndClassStart + @"
+        private readonly HashSet<string> MySet = new HashSet<string>();
+        private const string FieldItem = ""Item"";
+
+        public MyClass()
+        {
+            if (![|MySet.Contains(FieldItem)|])
+            {
+                MySet.Add(FieldItem);
+            }
+        }" + CSNamespaceAndClassEnd;
+
+            string fixedSource = CSUsings + CSNamespaceAndClassStart + @"
+        private readonly HashSet<string> MySet = new HashSet<string>();
+        private const string FieldItem = ""Item"";
+
+        public MyClass()
+        {
+            MySet.Add(FieldItem);
+        }" + CSNamespaceAndClassEnd;
+
+            await VerifyCS.VerifyCodeFixAsync(source, fixedSource);
+        }
+
+        [Fact]
+        public async Task ContainsAndAddCalledWithSameArgumentsLocals_OffersFixer_CS()
+        {
+            string source = CSUsings + CSNamespaceAndClassStart + @"
+        private readonly HashSet<string> MySet = new HashSet<string>();
+
+        public MyClass()
+        {
+            const string LocalItem = ""Item"";
+
+            if (![|MySet.Contains(LocalItem)|])
+            {
+                MySet.Add(LocalItem);
+            }
+        }" + CSNamespaceAndClassEnd;
```

Comment:
💡 It would be preferable to use raw strings instead of verbatim strings for all of these. If you only convert a subset, I would suggest starting with cases that are not parameterized.

Reply:
Changed in [6f69703](https://github.com/dotnet/roslyn-analyzers/pull/6767/commits/6f6970341a2f77e921f92ec32e6d050fa7bf780c).